test/extended/builds/build_pruning.go

```diff
@@ -47,21 +48,28 @@ var _ = g.Describe("[builds][pruning] prune builds based on settings in the buil
 		o.Expect(err).NotTo(o.HaveOccurred())
 
 		g.By("starting three test builds")
-		br, _ := exutil.StartBuildAndWait(oc, "myphp")
-		br.AssertSuccess()
-		br, _ = exutil.StartBuildAndWait(oc, "myphp")
-		br.AssertSuccess()
-		br, _ = exutil.StartBuildAndWait(oc, "myphp")
-		br.AssertSuccess()
+		for i := 0; i < 3; i++ {
+			br, _ := exutil.StartBuildAndWait(oc, "myphp")
+			br.AssertSuccess()
+		}
 
 		buildConfig, err := oc.Client().BuildConfigs(oc.Namespace()).Get("myphp", metav1.GetOptions{})
 		if err != nil {
 			fmt.Fprintf(g.GinkgoWriter, "%v", err)
 		}
 
-		builds, err := oc.Client().Builds(oc.Namespace()).List(metav1.ListOptions{})
-		if err != nil {
-			fmt.Fprintf(g.GinkgoWriter, "%v", err)
+		var builds *buildapi.BuildList
+
+		g.By("waiting up to five seconds for pruning to complete")
```

Comment:
Please use the `wait.PollImediate` method and you should probably wait a least a minute since you're able to detect success and finish early.  The e2e machines are heavily loaded and a flake on time is something we don't need.